src/containers/guest-home-page/signup-dialog/SignUpDialog.jsx

```
@@ -13,12 +13,21 @@ import studentImg from '~/assets/img/signup-dialog/student.svg'
+import { useSignUpMutation } from '~/services/auth-service'
+
 const SignUpDialog = ({ actionType }) => {
   const { t } = useTranslation()
+  const [signUpUser] = useSignUpMutation()
 
   const { handleSubmit, handleInputChange, handleBlur, data, errors } = useForm(
     {
-      onSubmit: async () => {},
+      onSubmit: async () => {
+        try {
+          const res = await signUpUser({ role: actionType, ...data })
```

ESLint

[error] 'res' is assigned a value but never used.

(no-unused-vars)